src/vs/workbench/contrib/comments/browser/commentsAccessibleView.ts

```diff
@@ -56,30 +57,31 @@ export class CommentsAccessibleView extends Disposable implements IAccessibleVie
 					}
 				};
 			});
-			return {
-				id: AccessibleViewProviderId.Notification,
-				provideContent: () => {
-					return content;
-				},
-				onClose(): void {
-					commentsView.focus();
-				},
-				next(): void {
+			return new AccessibleContentProvider(
+				AccessibleViewProviderId.Notification,
+				{ type: AccessibleViewType.View },
+				() => content,
+				() => commentsView.focus(),
+				AccessibilityVerbositySettingId.Comments,
+				undefined,
+				actions,
+				() => {
 					commentsView.focus();
 					commentsView.focusNextNode();
 					resolveProvider();
 				},
-				previous(): void {
+				() => {
 					commentsView.focus();
 					commentsView.focusPreviousNode();
 					resolveProvider();
-				},
-				verbositySettingKey: AccessibilityVerbositySettingId.Comments,
-				options: { type: AccessibleViewType.View },
-				actions
-			};
+				}
+			);
+		}
+		this._provider = resolveProvider();
+		if (this._provider) {
+			this._register(this._provider);
```

Comment:
I think you might have the same leak here, or did you mean to say `this._provider ??= resolveProvider()`

Reply:
whoops, missed this case, thanks